scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -658,6 +658,16 @@ protected boolean createExecution(NodeSet nodeSet, Node node, InternalJob job, I
 
                     tlogger.debug(task.getId(), "deploying");
 
+                    // Dynamically adjust the start-task-timeout according to the number dependency tasks in a merge.
+                    // above 500 parent tasks, it is worth adjusting.
+                    if (taskDescriptor.getParents().size() > 500) {
+                        DOTASK_ACTION_TIMEOUT = (int) (taskDescriptor.getParents().size() / 500.0 *
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)

Reply:
This is legacy code and cannot simply we cannot simply transform the method into static

Reply:
It saying that static field should be touched only by static methods. So maybe you can make DOTAST_ACTION_TIMEOUT non-static. Otherwise, you can ignore this by putting     @SuppressWarnings(" squid:S2696") to ignore rule here (then you have to justify this supressing in comments to the annotation). And, at least, make sure that there is no concurrency issue, when two thread call createExecution method.

Reply:
Why don't you use a local variable instead of modifying the static one ?